### PR TITLE
Feature/make tests optional

### LINKED ISSUE #111: [Feature] make tests optional
in my current project, we do snapshot tests with https://github.com/storybooks/storyshots
and acceptance-tests (with chimp).

We therefore forgo typical unit tests of components.

it would be cool to be able to configure whether these tests should be created or not

### PR DESCRIPTION
**Pull request by macrozone**
fixes https://github.com/mantrajs/mantra-cli/issues/111

sorry, there is no test yet for this feature ;-) (so ironic)

### CODE REVIEW

File: lib/generators/component.js [hunk 1]
```diff
@@ -12,19 +12,25 @@ export function generateComponent(name, options, config) {
...
   let [moduleName, entityName] = name.split(':');
 
   ensureModuleNameProvided(name);
   ensureModuleExists(moduleName);
 
   removeFile(getOutputPath('component', entityName, moduleName));
   destroyStorybook(name);
-  removeFile(getTestOutputPath('component', entityName, moduleName));
+
+  if(config.generateComponentTests) {
```
Comment (sungwoncho) [inline lib/generators/component.js]: not sure if the conditional is needed. I think we should always delete everything related to the component when removing.

File: lib/generators/container.js [hunk 2]
```diff
@@ -12,22 +12,32 @@ export function generateContainer(name, options, config) {
...
   }
 }
 
-export function destroyContainer(name) {
+export function destroyContainer(name, options, config) {
   let [moduleName, entityName] = name.split(':');
 
   removeFile(getOutputPath('container', entityName, moduleName));
-  removeFile(getTestOutputPath('container', entityName, moduleName));
+  if(config.generateContainerTests) {
```
Comment (sungwoncho) [inline lib/generators/container.js]: please see https://github.com/mantrajs/mantra-cli/pull/112/files#r100253160

**Review by sungwoncho — CHANGES_REQUESTED** [review]

**Comment by sungwoncho** [comment]
left a comment. open to suggestions.

File: lib/generators/component.js (hunk 1 above)
Comment (macrozone) [reply lib/generators/component.js]: Was not sure about this. I thought, well if someone disabled test-creation and added some tests as well, that would be none of our business ;-)

but after second thought, you are right. I will remove that condition

**Comment by macrozone** [comment]
@sungwoncho ok changed it!

**Comment by sungwoncho** [comment]
It seems that test is failing. Please see https://travis-ci.org/mantrajs/mantra-cli/jobs/199922989#L2937-L2956. Maybe something broke in this branch.

**Comment by macrozone** [comment]
ok tests are fixed. also added cases for the new configs

**Comment by macrozone** [comment]
@sungwoncho can you take a look?

**Comment by sungwoncho** [comment]
I'll look and merge this weekend

**Comment by sungwoncho** [comment]
LGTM. You can merge and publish. Don't forget to update https://github.com/mantrajs/mantra-cli/blob/master/RELEASE_NOTE.md. Thanks for your work.

**Comment by macrozone** [comment]
I will do the release, but i can't merge here ;-)

**Review by sungwoncho — APPROVED** [review]

**Comment by macrozone** [comment]
ok release is done, but i can't add a release tag here

**Comment by sungwoncho** [comment]
@macrozone I thought we were both admins to this repo. Could you check https://github.com/mantrajs/mantra-cli/invitations?

**Comment by sungwoncho** [comment]
Merged and tagged the release.

Thanks for your hard work. Please check your admin invitation.

**Pull request merged**